File: sorting/Python/sort_an_array_insert.py
class Solution:
    def sortArray(self, nums: list[int]) -> list[int]:
        res = nums.copy()
        for i in range(len(nums)):
            j = i-1
            if j <= 0 and res[j] > res[j+1]:
                temp = res[j]
                res[j] = res[j+1]
                res[j+1] = temp
                j -= 1
        return res

File: sorting/C_Cpp/sort_an_array_insert.c
#include <stdlib.h>

int* sortArray(int* nums, int numsSize, int* returnSize) {
    int n = numsSize;
    int *res = (int*)malloc(sizeof(int)*n);


    for (int i=0; i<n; i++) {
        res[i] = nums[i];
    }

    for (int i=1; i<n; i++) {
        int j = i-1;
        while (j>=0 && res[j+1] < res[j]) {
            int temp = res[j+1];
            res[j+1] = res[j];
            res[j] = temp;
            j-=1;
        }
    }

    *returnSize = n;
    return res;
}
